CMakeLists.txt

```diff
@@ -74,6 +74,32 @@ IF (TNTCXX_BUILD_TESTING)
     CONFIGURE_FILE(./test/gen_ssl.sh test_gen_ssl.sh COPYONLY)
 ENDIF()
 
+OPTION(TNTCXX_ENABLE_SANITIZERS
+    "If ON, tntcxx will be instrumented with sanitizers."
+    OFF)
+
+IF (TNTCXX_ENABLE_SANITIZERS)
+    SET(SANITIZER_FLAGS "-fsanitize=address")
+    # FIXME(gh-62)
+    IF (NOT CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+        # Leak sanitizer is not available in AppleClang, see
+        # google/sanitizers#1026 for details.
+        SET(SANITIZER_FLAGS ${SANITIZER_FLAGS} "-fsanitize=leak")
+    ENDIF()
+    IF (NOT CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+        # See https://gcc.gnu.org/bugzilla/show_bug.cgi?id=67762 for details.
+        SET(SANITIZER_FLAGS ${SANITIZER_FLAGS} "-fsanitize=undefined")
+        # Memory sanitizer is also not available in GCC.
+        IF (NOT CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+            # Memory sanitizer is not available on macOS, see
+            # google/sanitizers#841 for details.
+            SET(SANITIZER_FLAGS ${SANITIZER_FLAGS} "-fsanitize=memory")
+        ENDIF()
```

Comment:
What's about GCC on macOS?

Reply:
Not sure what you were talking about. Originally, this branch was meant for clang.

I dropped the memory sanitizer, since it requires instrumenting all dependencies, including libc and libc++, while it's just provides a subset of valgrind's functionality:
> It is critical that you should build all the code in your program (including libraries it uses, in particular, C++ standard library) with MSan. See [MemorySanitizerLibcxxHowTo](https://github.com/google/sanitizers/wiki/MemorySanitizerLibcxxHowTo) for more details. [^1]

[^1]: https://github.com/google/sanitizers/wiki/MemorySanitizer#using-instrumented-libraries